pkg/storage/indexheader/index/postings.go

```diff
@@ -234,18 +238,34 @@ func NewPostingOffsetTableFromSparseHeader(factory *streamencoding.DecbufFactory
 		postingOffsetsInMemSampling: postingOffsetsInMemSampling,
 	}
 
-	for sName, sOffsets := range postingsOffsetTable.Postings {
-		t.postings[sName] = &postingValueOffsets{
-			offsets: make([]postingOffset, len(sOffsets.Offsets)),
+	var step int
+	var ok bool
+	pbSampling := int(postingsOffsetTable.GetPostingOffsetInMemorySampling())
+	if (pbSampling > 0) && (pbSampling <= postingOffsetsInMemSampling) {
+		// if the sampling rate in the sparse index-header is set lower (more frequent) than
+		// the configured postingOffsetsInMemSampling we downsample to the configured rate
+		step, ok = stepSize(pbSampling, postingOffsetsInMemSampling)
+		if !ok {
+			return nil, fmt.Errorf("sparse index-header sampling rate not compatible with in-mem-sampling rate")
 		}
+	} else if (pbSampling > 0) && (pbSampling > postingOffsetsInMemSampling) {
+		// if the sparse index-header sampling rate is set higher must reconstruct from index-header
+		return nil, fmt.Errorf("sparse index-header sampling rate exceeds in-mem-sampling rate")
+	}
 
+	for sName, sOffsets := range postingsOffsetTable.Postings {
```

Comment:
what happens if the proto file didn't contain offset sampling? don't we get a panic? I'm thinking about the case when a new store-gateway with this change starts against a sparse header written by an old store-gateway without this change 

i think we need a unit test for this too